src/hivpy/demographics.py

```diff
@@ -193,3 +229,27 @@ def initialise_age(self, count):
             age_distribution = ContinuousAgeDistribution.select_model(self.params['inc_cat'])
 
         return age_distribution.gen_ages(count)
+
+    def determine_deaths(self, population_data: pd.DataFrame) -> pd.Series:
+        """Get which individuals die in a time step, as a boolean Series."""
+        all_died = pd.Series([False] * len(population_data))
+        for sex in SexType:
+            for (low_age, high_age), rate in self.params["death_rates"][sex].items():
+                # Find everyone in the relevant age range
+                index = selector(population_data, sex=(operator.eq, sex),
+                                 age=(between, (low_age, high_age)))
+                group = population_data[index]
+                # Probability of dying, assuming time step of 3 months
+                prob_of_death = 1 - exp(-rate / 4)
+                died = np.random.choice([True, False], size=len(group),
+                                        p=[prob_of_death, 1 - prob_of_death])
+                # Mark those who died from this group
+                all_died |= pd.Series(died, index=group.index)
+        return population_data.date_of_death.isnull() & all_died
```

Comment:
Why not incorporate the condition that they be not already dead into the selector so we don't need to calculate the probabilities for dead people?